mtwitter.js

```diff
@@ -0,0 +1,254 @@
+var Twitter = require('twitter');
+var Request = require('request');
+var fs = require('fs');
+var Buffer = require('buffer').Buffer;
+
+var TWITTER_PROFILE_INTERVAL = 60000;
+
+var mtwitter = function(bridge, config) {
+  this.app_auth = config.app_auth;
+  this.app_twitter = null;
+  this.tuser_cache = {};
+  this.timeline_list = [];
+  this.timeline_queue = [];
+  this.timeline_period = 0;
+  this.timeline_intervalobj = null;
+  this._bridge = bridge;
+
+  this.get_bearer_token((bt) => {
+      if (bt != null) {
+          this.app_auth.bearer_token = bt;
+          console.log("Twitter Application Auth OK!");
+          this.app_twitter = new Twitter(this.app_auth);
+          return;
+      }
+      console.error("Twitter Application Failed Auth. The bridge will be operating in a limited capacity.");
+  });
+}
+
+mtwitter.prototype.get_bearer_token = function(cb){
+  try {
+      fs.accessSync('bearer.tok', fs.R_OK);
+      var tok = fs.readFileSync('bearer.tok', 'utf8');
+      cb(tok);
```

Comment:
How do you know this token hasn't been revoked? Might it be worth checking to see if the token is still valid before giving this to the caller, or do you expect the caller to assert validity?